datacube/utils/generic.py

```diff
@@ -26,3 +39,64 @@ def map_with_lookahead(it, if_one=None, if_many=None):
 
     for v in itertools.chain(iter(p1), it):
         yield proc(v)
+
+
+def qmap(func, q, eos_marker=EOS):
+    """ Converts queue to an iterator.
+
+    For every `item` in the `q` that is not `eos_marker`, `yield proc(item)`
+
+    Takes care of calling `.task_done()` on every item extracted from the queue.
+    """
+    while True:
+        item = q.get(block=True)
+        if item is eos_marker:
+            q.task_done()
```

Comment:
Is `.task_done()` a dask thing?

`qmap()` and `it2q` are only used in `compute_map()`, and for `qmap` at least don't seem that useful elsewhere. They're adding functionality on top of a standard `queue`, so might be clearer as a new type, instead of extra functions.

Or maybe just moving them into the `dask` module. Not sure, just flagging my confusion.

Reply:
It's python queue thing, like `thread.join()` but on a queue.

Reply:
basically if you want to use `queue.join()` you need to call `.task_done`, so I'm doing it here in case callers do use that functionality

Reply:
So those things are useful for multi-threaded code, not necessarily dask code. Basically it allows one to write code that just stitches a data processing pipeline from code that consumes and produces iterators, but uses a separate thread for every stage of the processing pipeline.

I'm using that in more places in `odc-tools`, particularly in indexing pipelines, and hoping to bring more of that code into core.